Sector Delta Popup

SahanYW opened the pull request:
Creates a popup every sector displaying time difference to personal best lap.

Wesley1041 commented on src/StarterGui/WesTimingGui/TimingLocalScript/TimingKeeperService.lua:
```diff
@@ -40,7 +40,35 @@ function service:HandleSector(sector: number)
+	end
+	
+	-- Get delta time data
+	if _config.SectorTimeDelta == true then
```
`== true` is redundant.

```suggestion
	if _config.SectorTimeDelta then
```

Wesley1041 commented on src/Workspace/Wes_Timing_System_4/_Config.lua:
Can you revert the changes in this file?

Wesley1041 commented on src/StarterGui/WesTimingGui/TimingLocalScript/TimingKeeperService.lua:
```diff
@@ -40,7 +40,35 @@ function service:HandleSector(sector: number)
+
+			-- Improved sector
+			if deltaTimeData.delta <= 0 then
+				displayString = "- " .. displayString
```
I prefer no whitespace between `-` and the digits
```suggestion
				displayString = "-" .. displayString
```

Wesley1041 commented on src/StarterGui/WesTimingGui/TimingLocalScript/TimingKeeperService.lua:
```diff
@@ -40,7 +40,35 @@ function service:HandleSector(sector: number)
+
+				-- Slower sector
+			else
+				displayString = "+ " .. displayString
```
```suggestion
				displayString = "+" .. displayString
```

Wesley1041 commented on src/Workspace/Wes_Timing_System_4/_Config.lua:
```diff
@@ -17,11 +17,13 @@ local config = {
-	PopupEverycut = false,
+	PopupEverycut = true,
+	-- Shows time difference to personal best lap every sector
+	SectorTimeDelta = true,
```
I think the default value should be false because I think the timing system should be relatively basic at its core.
```suggestion
	SectorTimeDelta = false,
```

Wesley1041 reviewed (changes requested):
Thanks for the effort! I have added a few comments though, which should be addressed first before completing this PR.

I mainly think that the new feature should be optional and not be enabled by default. One potential thing that may turn people away from the feature is that the displayed deltas are based on the individual sectors. I think most drivers expect the sector 2 popup to display the delta between S1 + S2 instead of just S2. I think it would be nice to implement that version of this feature in the future.

SahanYW commented:
Just fixed the deltas only showing sector deltas instead of S1 + S2. I didn't even realize that I made that mistake even though it was my original intention.
Also fixed config and the other minor problems.
